src/blockdata/transaction.rs

```diff
@@ -201,6 +210,14 @@ pub struct OwnedTxOut<'a> {
     pub sub_index: Index,
     /// The associated transaction public key.
     pub tx_pubkey: PublicKey,
+    /// The unblinded amount of this output.
+    ///
+    /// None if we didn't have enough information to unblind the output.
+    pub amount: Option<u64>,
+    /// The original blinding factor of this output.
+    ///
+    /// None if we didn't have enough information to unblind the output.
+    pub blinding_factor: Option<Scalar>,
```

Comment:
:upside_down_face: It's somewhat annoying that either both of these will be present or both will be missing.

Reply:
Should we model it as a dedicated struct? It is a bit annoying having too many layers here, so maybe a dedicated struct and getters for the individual fields that pass through?

Any other ideas?

Reply:
I like what you suggest!